### ISSUE TITLE
aliyundrive@6.0.0: download via aria2 failed

### ISSUE THREAD

**Issue by cscnk52** [chawyehsu/dorado#865]
ERROR Download failed! (Error 22) HTTP response header was bad or unexpected
ERROR https://cdn.aliyundrive.net/downloads/apps/desktop/update/6.0.0/win32/ia32/aDrive-6.0.0.exe#/aDrive-6.0.0.7z
    referer=https://cdn.aliyundrive.net/downloads/apps/desktop/update/6.0.0/win32/ia32/aDrive-6.0.0.exe#/
    dir=C:\scoop\cache
    out=aliyundrive#6.0.0#67baabb.7z

ERROR & 'C:\scoop\apps\aria2\current\aria2c.exe' --input-file='C:\scoop\cache\aliyundrive.txt' --user-agent='Scoop/1.0 (+http://scoop.sh/) PowerShell/7.4 (Windows NT 10.0; Win64; x64; Core)' --allow-overwrite=true --auto-file-renaming=false --retry-wait=2 --split=5 --max-connection-per-server=5 --min-split-size=5M --console-log-level=warn --enable-color=false --no-conf=true --follow-metalink=true --metalink-preferred-protocol=https --min-tls-version=TLSv1.2 --stop-with-process=14712 --continue --summary-interval=0 --auto-save-interval=1

Please try again or create a new issue by using the following link and paste your console output:
https://github.com/chawyehsu/dorado/issues/new?title=aliyundrive%406.0.0%3a+download+via+aria2+failed

**Comment by socratesng** [comment]
For the latest version in the bucket (6.3.2), the correct installation file can be download from
[https://cdn.aliyundrive.net/downloads/apps/desktop/aDrive-6.3.2.exe]
instead of 
[https://cdn.aliyundrive.net/downloads/apps/desktop/update/6.3.2/win32/ia32/aDrive-6.3.2.exe]

It seems reasonable to remove "/update/$version/win32/ia32" from the "url" properties in the manifest.

**Comment by cscnk52** [comment]
absolute yes！
I encountered this issue a few days ago as well, and I'm currently working on maintaining my own manifest for AliyunDrive. However, it seems there's no effective way to handle the blocking caused by the referer.

https://github.com/cscnk52/cetacea/blob/7cafeeabc94f72aa73be66990391ff8e70687370/bucket/aliyundrive.json
https://github.com/cscnk52/cetacea/issues/6 (in chinese)

or you can trying to making a PR to fix this issue, enjoy!

**Comment by socratesng** [comment]
I'm not familiar with scoop or web. Can't help with fixing the issue, T_T

But by checking the existing manifest, it seems refer is not needed, which is also evident when I just paste and open the download link in another browser successfully.

**Comment by cscnk52** [comment]
## TL;DR
after download `aDrive.exe` from official website, if you scoop's aria2 is enabled, you can find the download file name in log, just simply rename `aDrive.exe` to `aliyundrive#6.3.3#80771de.7z`(shown as picture), and place file in `$SCOOP/cache`, copying following code, and enjoy !

![image](https://github.com/user-attachments/assets/08a2d6d4-d42f-4c1f-9f42-0050f55ac7c7)

```
scoop bucket add cetacea https://github.com/cscnk52/cetacea
scoop install cetacea/aliyundrive
```

## why?

because as what i said, the cdn of aliyundrive forbidden downloading with referer(See https://github.com/cscnk52/cetacea/issues/6), but seem like scoop's download only support download with referer, i am work on this issue, after fix this issue, i will making a PR to this repo. and seem like checkver of dorado/aliyundrive is not timely, i will fix it in below PR too.

**Comment by socratesng** [comment]
I see. Many thanks for the ad-hoc solution.

I actually need to disable aria2 to successfully (directly) load the 7z file from the cache.
